### PR TITLE
refactor(vpc): move secgroup and secgroup rule to services

### PR DESCRIPTION
**Pull request by Jason-Zhang9309**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/vpc' TESTARGS='-run TestAccNetworkingV3SecGroup'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/vpc -v -run TestAccNetworkingV3SecGroup -timeout 360m -parallel 4
=== RUN   TestAccNetworkingV3SecGroup_basic
=== PAUSE TestAccNetworkingV3SecGroup_basic
=== RUN   TestAccNetworkingV3SecGroup_withEpsId
=== PAUSE TestAccNetworkingV3SecGroup_withEpsId
=== RUN   TestAccNetworkingV3SecGroup_noDefaultRules
=== PAUSE TestAccNetworkingV3SecGroup_noDefaultRules
=== CONT  TestAccNetworkingV3SecGroup_basic
=== CONT  TestAccNetworkingV3SecGroup_noDefaultRules
=== CONT  TestAccNetworkingV3SecGroup_withEpsId
    acceptance.go:305: The environment variables does not support Enterprise Project ID for acc tests
--- SKIP: TestAccNetworkingV3SecGroup_withEpsId (0.00s)
--- PASS: TestAccNetworkingV3SecGroup_noDefaultRules (20.71s)
--- PASS: TestAccNetworkingV3SecGroup_basic (29.02s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/vpc       29.071s

make testacc TEST='./huaweicloud/services/acceptance/vpc' TESTARGS='-run TestAccNetworkingSecGroupRule'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/vpc -v -run TestAccNetworkingSecGroupRule -timeout 360m -parallel 4
=== RUN   TestAccNetworkingSecGroupRule_basic
=== PAUSE TestAccNetworkingSecGroupRule_basic
=== RUN   TestAccNetworkingSecGroupRule_oldPorts
=== PAUSE TestAccNetworkingSecGroupRule_oldPorts
=== RUN   TestAccNetworkingSecGroupRule_remoteGroup
=== PAUSE TestAccNetworkingSecGroupRule_remoteGroup
=== RUN   TestAccNetworkingSecGroupRule_lowerCaseCIDR
=== PAUSE TestAccNetworkingSecGroupRule_lowerCaseCIDR
=== RUN   TestAccNetworkingSecGroupRule_noPorts
=== PAUSE TestAccNetworkingSecGroupRule_noPorts
=== RUN   TestAccNetworkingSecGroupRule_remoteAddressGroup
=== PAUSE TestAccNetworkingSecGroupRule_remoteAddressGroup
=== RUN   TestAccNetworkingSecGroupRule_action
=== PAUSE TestAccNetworkingSecGroupRule_action
=== RUN   TestAccNetworkingSecGroupRule_priority
=== PAUSE TestAccNetworkingSecGroupRule_priority
=== CONT  TestAccNetworkingSecGroupRule_basic
=== CONT  TestAccNetworkingSecGroupRule_noPorts
=== CONT  TestAccNetworkingSecGroupRule_remoteGroup
=== CONT  TestAccNetworkingSecGroupRule_oldPorts
--- PASS: TestAccNetworkingSecGroupRule_basic (34.98s)
=== CONT  TestAccNetworkingSecGroupRule_lowerCaseCIDR
--- PASS: TestAccNetworkingSecGroupRule_remoteGroup (35.58s)
=== CONT  TestAccNetworkingSecGroupRule_action
--- PASS: TestAccNetworkingSecGroupRule_oldPorts (35.93s)
=== CONT  TestAccNetworkingSecGroupRule_priority
--- PASS: TestAccNetworkingSecGroupRule_noPorts (36.08s)
=== CONT  TestAccNetworkingSecGroupRule_remoteAddressGroup
--- PASS: TestAccNetworkingSecGroupRule_lowerCaseCIDR (34.31s)
--- PASS: TestAccNetworkingSecGroupRule_priority (33.85s)
--- PASS: TestAccNetworkingSecGroupRule_remoteAddressGroup (33.93s)
--- PASS: TestAccNetworkingSecGroupRule_action (35.79s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/vpc       71.441s

make testacc TEST='./huaweicloud/services/acceptance/vpc' TESTARGS='-run TestAccNetworkingSecGroupV3DataSource'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/vpc -v -run TestAccNetworkingSecGroupV3DataSource -timeout 360m -parallel 4
=== RUN   TestAccNetworkingSecGroupV3DataSource_basic
=== PAUSE TestAccNetworkingSecGroupV3DataSource_basic
=== RUN   TestAccNetworkingSecGroupV3DataSource_byID
=== PAUSE TestAccNetworkingSecGroupV3DataSource_byID
=== CONT  TestAccNetworkingSecGroupV3DataSource_basic
=== CONT  TestAccNetworkingSecGroupV3DataSource_byID
--- PASS: TestAccNetworkingSecGroupV3DataSource_byID (41.82s)
--- PASS: TestAccNetworkingSecGroupV3DataSource_basic (42.38s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/vpc       42.441s
```

### CODE REVIEW

File: huaweicloud/services/acceptance/vpc/data_source_huaweicloud_networking_secgroup_test.go [hunk 1]
```diff
@@ -7,15 +7,16 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
 )
 
 func TestAccNetworkingSecGroupV3DataSource_basic(t *testing.T) {
 	var rName = fmt.Sprintf("tf-acc-test-%s", acctest.RandString(5))
```
Comment (hmbig2) [inline huaweicloud/services/acceptance/vpc/data_source_huaweicloud_networking_secgroup_test.go]:
```suggestion
	rName := acceptance.RandomAccResourceNameWithDash()
```

File: huaweicloud/services/acceptance/vpc/resource_huaweicloud_networking_secgroup_rule_test.go [hunk 2]
```diff
@@ -10,6 +10,7 @@ import (
 
 	"github.com/chnsz/golangsdk/openstack/networking/v1/security/rules"
 	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
 	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/fmtp"
```
Comment (hmbig2) [inline huaweicloud/services/acceptance/vpc/resource_huaweicloud_networking_secgroup_rule_test.go]: use fmt

File: huaweicloud/services/acceptance/vpc/resource_huaweicloud_networking_secgroup_rule_test.go [hunk 3]
```diff
@@ -166,9 +167,9 @@ func TestAccNetworkingSecGroupRule_remoteAddressGroup(t *testing.T) {
 	rName := fmt.Sprintf("tf-acc-test-%s", acctest.RandString(5))
```
Comment (hmbig2) [inline huaweicloud/services/acceptance/vpc/resource_huaweicloud_networking_secgroup_rule_test.go]: ditto

File: huaweicloud/services/acceptance/vpc/resource_huaweicloud_networking_secgroup_rule_test.go [hunk 4]
```diff
@@ -251,8 +252,8 @@ func TestAccNetworkingSecGroupRule_priority(t *testing.T) {
 }
 
 func testAccCheckNetworkingSecGroupRuleDestroy(s *terraform.State) error {
```
Comment (hmbig2) [inline huaweicloud/services/acceptance/vpc/resource_huaweicloud_networking_secgroup_rule_test.go]:
```suggestion
	rc := acceptance.InitResourceCheck(
		resourceName,
		&ar,
		xxx,
	)
```

replace checkDestroy and check exit with this function

File: huaweicloud/services/vpc/data_source_huaweicloud_networking_secgroup.go [hunk 5]
```diff
@@ -70,7 +70,7 @@ func getRuleListByGroupId(client *golangsdk.ServiceClient, groupId string) ([]ma
 
 func dataSourceNetworkingSecGroupRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	config := meta.(*config.Config)
-	region := GetRegion(d, config)
+	region := config.GetRegion(d)
 	v3Client, err := config.NetworkingV3Client(region)
 	if err != nil {
 		return fmtp.DiagErrorf("Error creating HuaweiCloud networking v3 client: %s", err)
```
Comment (hmbig2) [inline huaweicloud/services/vpc/data_source_huaweicloud_networking_secgroup.go]:
```suggestion
		return diag.DiagErrorf("error creating networking v3 client: %s", err)
```

File: huaweicloud/services/vpc/resource_huaweicloud_networking_secgroup.go [hunk 6]
```diff
@@ -141,17 +141,17 @@ func ResourceNetworkingSecGroup() *schema.Resource {
...
 	}
 
 	// Only name and enterprise project ID are supported.
 	createOpts := v3groups.CreateOpts{
 		Name:                d.Get("name").(string),
-		EnterpriseProjectId: GetEnterpriseProjectID(d, config),
+		EnterpriseProjectId: cfg.GetEnterpriseProjectID(d),
 	}
 
 	logp.Printf("[DEBUG] Create HuaweiCloud Security Group: %#v", createOpts)
```
Comment (hmbig2) [inline huaweicloud/services/vpc/resource_huaweicloud_networking_secgroup.go]:
```suggestion
	log.Printf("[DEBUG] Create HuaweiCloud Security Group: %#v", createOpts)
```

File: huaweicloud/services/vpc/resource_huaweicloud_networking_secgroup_rule.go [hunk 7]
```diff
@@ -239,14 +239,14 @@ func resourceNetworkingSecGroupRuleCreate(ctx context.Context, d *schema.Resourc
...
 
-	v1Client, err := config.NetworkingV1Client(region)
+	v1Client, err := cfg.NetworkingV1Client(region)
 	if err != nil {
 		return fmtp.DiagErrorf("Error creating HuaweiCloud networking v1 client: %s", err)
 	}
-	v3Client, err := config.NetworkingV3Client(GetRegion(d, config))
+	v3Client, err := cfg.NetworkingV3Client(cfg.GetRegion(d))
 	if err != nil {
 		return fmtp.DiagErrorf("Error creating HuaweiCloud networking v3 client: %s", err)
```
Comment (hmbig2) [inline huaweicloud/services/vpc/resource_huaweicloud_networking_secgroup_rule.go]: ditto

File: huaweicloud/services/acceptance/vpc/data_source_huaweicloud_networking_secgroup_test.go (hunk 1 above)
Comment (Jason-Zhang9309) [reply huaweicloud/services/acceptance/vpc/data_source_huaweicloud_networking_secgroup_test.go]: fixed

File: huaweicloud/services/acceptance/vpc/resource_huaweicloud_networking_secgroup_rule_test.go (hunk 2 above)
Comment (Jason-Zhang9309) [reply huaweicloud/services/acceptance/vpc/resource_huaweicloud_networking_secgroup_rule_test.go]: fixed

File: huaweicloud/services/acceptance/vpc/resource_huaweicloud_networking_secgroup_rule_test.go (hunk 3 above)
Comment (Jason-Zhang9309) [reply huaweicloud/services/acceptance/vpc/resource_huaweicloud_networking_secgroup_rule_test.go]: fixed

File: huaweicloud/services/acceptance/vpc/resource_huaweicloud_networking_secgroup_rule_test.go (hunk 4 above)
Comment (Jason-Zhang9309) [reply huaweicloud/services/acceptance/vpc/resource_huaweicloud_networking_secgroup_rule_test.go]: fixed

File: huaweicloud/services/vpc/data_source_huaweicloud_networking_secgroup.go (hunk 5 above)
Comment (Jason-Zhang9309) [reply huaweicloud/services/vpc/data_source_huaweicloud_networking_secgroup.go]: fixed

File: huaweicloud/services/vpc/resource_huaweicloud_networking_secgroup.go (hunk 6 above)
Comment (Jason-Zhang9309) [reply huaweicloud/services/vpc/resource_huaweicloud_networking_secgroup.go]: fixed

File: huaweicloud/services/vpc/resource_huaweicloud_networking_secgroup_rule.go (hunk 7 above)
Comment (Jason-Zhang9309) [reply huaweicloud/services/vpc/resource_huaweicloud_networking_secgroup_rule.go]: fixed

**Comment by Jason-Zhang9309** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/vpc' TESTARGS='-run TestAccNetworkingSecGroupV3DataSource'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/vpc -v -run TestAccNetworkingSecGroupV3DataSource -timeout 360m -parallel 4
=== RUN   TestAccNetworkingSecGroupV3DataSource_basic
=== PAUSE TestAccNetworkingSecGroupV3DataSource_basic
=== RUN   TestAccNetworkingSecGroupV3DataSource_byID
=== PAUSE TestAccNetworkingSecGroupV3DataSource_byID
=== CONT  TestAccNetworkingSecGroupV3DataSource_basic
=== CONT  TestAccNetworkingSecGroupV3DataSource_byID
--- PASS: TestAccNetworkingSecGroupV3DataSource_basic (43.57s)
--- PASS: TestAccNetworkingSecGroupV3DataSource_byID (43.79s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/vpc       43.851s

make testacc TEST='./huaweicloud/services/acceptance/vpc' TESTARGS='-run TestAccNetworkingV3SecGroup'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/vpc -v -run TestAccNetworkingV3SecGroup -timeout 360m -parallel 4
=== RUN   TestAccNetworkingV3SecGroup_basic
=== PAUSE TestAccNetworkingV3SecGroup_basic
=== RUN   TestAccNetworkingV3SecGroup_withEpsId
=== PAUSE TestAccNetworkingV3SecGroup_withEpsId
=== RUN   TestAccNetworkingV3SecGroup_noDefaultRules
=== PAUSE TestAccNetworkingV3SecGroup_noDefaultRules
=== CONT  TestAccNetworkingV3SecGroup_basic
=== CONT  TestAccNetworkingV3SecGroup_noDefaultRules
=== CONT  TestAccNetworkingV3SecGroup_withEpsId
    acceptance.go:305: The environment variables does not support Enterprise Project ID for acc tests
--- SKIP: TestAccNetworkingV3SecGroup_withEpsId (0.00s)
--- PASS: TestAccNetworkingV3SecGroup_noDefaultRules (19.69s)
--- PASS: TestAccNetworkingV3SecGroup_basic (28.84s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/vpc       28.950s

make testacc TEST='./huaweicloud/services/acceptance/vpc' TESTARGS='-run TestAccNetworkingSecGroupRule'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/vpc -v -run TestAccNetworkingSecGroupRule -timeout 360m -parallel 4
=== RUN   TestAccNetworkingSecGroupRule_basic
=== PAUSE TestAccNetworkingSecGroupRule_basic
=== RUN   TestAccNetworkingSecGroupRule_oldPorts
=== PAUSE TestAccNetworkingSecGroupRule_oldPorts
=== RUN   TestAccNetworkingSecGroupRule_remoteGroup
=== PAUSE TestAccNetworkingSecGroupRule_remoteGroup
=== RUN   TestAccNetworkingSecGroupRule_lowerCaseCIDR
=== PAUSE TestAccNetworkingSecGroupRule_lowerCaseCIDR
=== RUN   TestAccNetworkingSecGroupRule_noPorts
=== PAUSE TestAccNetworkingSecGroupRule_noPorts
=== RUN   TestAccNetworkingSecGroupRule_remoteAddressGroup
=== PAUSE TestAccNetworkingSecGroupRule_remoteAddressGroup
=== RUN   TestAccNetworkingSecGroupRule_action
=== PAUSE TestAccNetworkingSecGroupRule_action
=== RUN   TestAccNetworkingSecGroupRule_priority
=== PAUSE TestAccNetworkingSecGroupRule_priority
=== CONT  TestAccNetworkingSecGroupRule_basic
=== CONT  TestAccNetworkingSecGroupRule_noPorts
=== CONT  TestAccNetworkingSecGroupRule_action
=== CONT  TestAccNetworkingSecGroupRule_remoteAddressGroup
--- PASS: TestAccNetworkingSecGroupRule_basic (35.67s)
=== CONT  TestAccNetworkingSecGroupRule_priority
--- PASS: TestAccNetworkingSecGroupRule_noPorts (35.91s)
=== CONT  TestAccNetworkingSecGroupRule_remoteGroup
--- PASS: TestAccNetworkingSecGroupRule_remoteAddressGroup (36.12s)
=== CONT  TestAccNetworkingSecGroupRule_lowerCaseCIDR
--- PASS: TestAccNetworkingSecGroupRule_action (37.46s)
=== CONT  TestAccNetworkingSecGroupRule_oldPorts
--- PASS: TestAccNetworkingSecGroupRule_priority (34.45s)
--- PASS: TestAccNetworkingSecGroupRule_remoteGroup (35.33s)
--- PASS: TestAccNetworkingSecGroupRule_lowerCaseCIDR (35.28s)
--- PASS: TestAccNetworkingSecGroupRule_oldPorts (35.82s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/vpc       73.332s
```

**Pull request merged**